param/__init__.py

```diff
@@ -1201,6 +1203,176 @@ def get_range(self):
         raise NotImplementedError("get_range() must be implemented in subclasses.")
 
 
+
+class SelectObjects(list):
```

Comment:
This approach makes sense to me given the constraints, but I think the name doesn't reflect what is really happening here. The name that would have made the approach clearer to me at a glance might be something like `ListProxy`.

Reply:
Agree.